From cf11cb15dd958bbbcfa8d305065ec4e3f2109d9e Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Fri, 18 Oct 2019 09:37:37 +0800
Subject: [PATCH] ARM: dts: rockchip: remove the regulator-initial-mode for
 vcc_ddr

The driver does not match the dts, pmic driver is update to 4.19 kernel,
but the dts node is not update,make the vcc_ddr regulator init failed:
[    2.126351] rk808 0-001a: chip id: 0x8160
[    2.134083] rk808-regulator rk808-regulator: there is no dvs0 gpio
[    2.134229] rk808-regulator rk808-regulator: there is no dvs1 gpio
[    2.134347] DCDC_REG1: supplied by vcc_sys
[    2.136769] DCDC_REG2: supplied by vcc_sys
[    2.138759] DCDC_REG3: supplied by vcc_sys
[    2.139064] vcc_ddr: no set_mode operation
[    2.139110] rk808 0-001a: failed to register 2 regulator
[    2.140110] rk808-regulator: probe of rk808-regulator failed with error -22

Change-Id: Ieecd8678345202db0be5ac4c669c890283793d42
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 arch/arm/boot/dts/rk3036-echo.dts                  | 1 -
 arch/arm/boot/dts/rk3126-bnd-d708.dtsi             | 1 -
 arch/arm/boot/dts/rk3126-bnd-m88-emmc.dts          | 1 -
 arch/arm/boot/dts/rk3229-at-common.dtsi            | 1 -
 arch/arm/boot/dts/rk3229-echo-v10.dts              | 1 -
 arch/arm/boot/dts/rk3229-gva-sdk.dts               | 1 -
 arch/arm/boot/dts/rk3308-dot-rk816-v10-aarch32.dts | 1 -
 7 files changed, 7 deletions(-)

diff --git a/arch/arm/boot/dts/rk3036-echo.dts b/arch/arm/boot/dts/rk3036-echo.dts
index ffa608bb90a2..ab27a9c368c2 100644
--- a/arch/arm/boot/dts/rk3036-echo.dts
+++ b/arch/arm/boot/dts/rk3036-echo.dts
@@ -232,7 +232,6 @@
 
 			vcc_ddr: DCDC_REG3 {
 				regulator-name = "vcc_ddr";
-				regulator-initial-mode = <0x1>;
 				regulator-boot-on;
 				regulator-always-on;
 				regulator-state-mem {
diff --git a/arch/arm/boot/dts/rk3126-bnd-d708.dtsi b/arch/arm/boot/dts/rk3126-bnd-d708.dtsi
index fd21470f0ed6..e5501dff2b6b 100644
--- a/arch/arm/boot/dts/rk3126-bnd-d708.dtsi
+++ b/arch/arm/boot/dts/rk3126-bnd-d708.dtsi
@@ -409,7 +409,6 @@
 				regulator-name = "vcc_ddr";
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-initial-mode = <1>;
 			};
 
 			vcc_io: DCDC_REG4 {
diff --git a/arch/arm/boot/dts/rk3126-bnd-m88-emmc.dts b/arch/arm/boot/dts/rk3126-bnd-m88-emmc.dts
index a568177de348..427eaba16e72 100644
--- a/arch/arm/boot/dts/rk3126-bnd-m88-emmc.dts
+++ b/arch/arm/boot/dts/rk3126-bnd-m88-emmc.dts
@@ -352,7 +352,6 @@
 				regulator-name = "vcc_ddr";
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-initial-mode = <1>;
 			};
 
 			vcc_io: DCDC_REG4 {
diff --git a/arch/arm/boot/dts/rk3229-at-common.dtsi b/arch/arm/boot/dts/rk3229-at-common.dtsi
index f9b1db836710..147c413143fb 100644
--- a/arch/arm/boot/dts/rk3229-at-common.dtsi
+++ b/arch/arm/boot/dts/rk3229-at-common.dtsi
@@ -175,7 +175,6 @@
 
 			vcc_ddr: DCDC_REG3 {
 				regulator-name = "vcc_ddr";
-				regulator-initial-mode = <0x1>;
 				regulator-boot-on;
 				regulator-always-on;
 				regulator-state-mem {
diff --git a/arch/arm/boot/dts/rk3229-echo-v10.dts b/arch/arm/boot/dts/rk3229-echo-v10.dts
index a319ceda797c..7d3e9b048020 100644
--- a/arch/arm/boot/dts/rk3229-echo-v10.dts
+++ b/arch/arm/boot/dts/rk3229-echo-v10.dts
@@ -241,7 +241,6 @@
 
 			vcc_ddr: DCDC_REG3 {
 				regulator-name = "vcc_ddr";
-				regulator-initial-mode = <0x1>;
 				regulator-boot-on;
 				regulator-always-on;
 				regulator-state-mem {
diff --git a/arch/arm/boot/dts/rk3229-gva-sdk.dts b/arch/arm/boot/dts/rk3229-gva-sdk.dts
index 6c76bc0da164..8c25d400c544 100644
--- a/arch/arm/boot/dts/rk3229-gva-sdk.dts
+++ b/arch/arm/boot/dts/rk3229-gva-sdk.dts
@@ -260,7 +260,6 @@
 
 			vcc_ddr: DCDC_REG3 {
 				regulator-name = "vcc_ddr";
-				regulator-initial-mode = <0x1>;
 				regulator-boot-on;
 				regulator-always-on;
 				regulator-state-mem {
diff --git a/arch/arm/boot/dts/rk3308-dot-rk816-v10-aarch32.dts b/arch/arm/boot/dts/rk3308-dot-rk816-v10-aarch32.dts
index d17ed78e628c..7de1dc209814 100644
--- a/arch/arm/boot/dts/rk3308-dot-rk816-v10-aarch32.dts
+++ b/arch/arm/boot/dts/rk3308-dot-rk816-v10-aarch32.dts
@@ -262,7 +262,6 @@
 				regulator-name = "vcc_ddr";
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-initial-mode = <1>;
 			};
 
 			vcc_io: DCDC_REG4 {
-- 
2.35.3

